index.js

```diff
@@ -109,23 +126,29 @@ Fireproof._handleError = function(onComplete) {
 /**
  * Tell Fireproof to use a given promise library from now on.
  * @method Fireproof.bless
- * @param {Q} Q a Q-style promise constructor with at least defer().
- * @throws if you don't provide a valid Deferred-style promise library.
+ * @param {Q} Q a Q-style promise constructor with an optional defer().
+ * @throws if you don't provide a valid promise library.
  */
 Fireproof.bless = function(newQ) {
 
-  if (newQ === undefined || newQ === null || typeof(newQ.defer) !== 'function') {
-    throw new Error('You tried to give Fireproof an invalid Q library!');
-  }
-
-  var deferred = newQ.defer();
+  function assert(value) {if (!value) throw null}
 
-  if (deferred === undefined || deferred === null ||
-    deferred.promise === undefined || deferred.promise === null ||
-    typeof(deferred.reject) !== 'function' ||
-    typeof(deferred.resolve) !== 'function' ||
-    typeof(deferred.promise.then) !== 'function') {
-    throw new Error('You tried to give Fireproof an invalid Q library!');
+  try {
```

Comment:
no need to wrap all this in a `try` block -- just raise the exception from right inside `assert`